pkg/proxy/iptables/proxier.go

```diff
@@ -571,143 +575,154 @@ func (proxier *Proxier) buildEndpointInfoList(endPoints []hostPortInfo, endpoint
 	for _, hpp := range endPoints {
 		key := net.JoinHostPort(hpp.host, strconv.Itoa(hpp.port))
 		if lookupSet.Has(key) {
-			filteredEndpoints = append(filteredEndpoints, &endpointsInfo{ip: key, localEndpoint: hpp.localEndpoint})
+			filteredEndpoints = append(filteredEndpoints, &endpointsInfo{endpoint: key, isLocal: hpp.isLocal})
```

Comment:
Not sure it matters, but if the endpoint record has invalid addresses, this will still add the endpoint to the map.  Maybe filter those out before calling buildEndpointInfoList()?  Or if you feel that apiserver validation is sufficient, ignore this.

```
func addTestEndpointAddress(array []api.EndpointAddress, ip, hostname string, nodename *string) []api.EndpointAddress {
	addr := api.EndpointAddress{
		IP:       ip,
		Hostname: hostname,
		NodeName: nodename,
	}
	return append(array, addr)
}

		makeTestEndpoint("somewhere", "invalid-addresses", func(s *api.EndpointSubset) {
			s.Addresses = addTestEndpointAddress(s.Addresses, "", "foobar.com", &node1)
			s.Addresses = addTestEndpointAddress(s.Addresses, "", "foobar.com", nil)
			s.Ports = addTestEndpointPort(s.Ports, "first", "UDP", 4444)
			s.Ports = addTestEndpointPort(s.Ports, "second", "TCP", 5555)
		}),
```

This invalid endpoint will still be added to the returned new endpoints map, it probably shouldn't.